python/taichi/misc/gui.py

```diff
@@ -123,30 +123,53 @@ def line(self, begin, end, radius, color):
         self.canvas.path_single(begin[0], begin[1], end[0], end[1], color,
                                 radius)
 
-    def show(self, file=None):
+    def show(self, file_or_image=None):
         self.core.update()
-        if file:
-            self.core.screenshot(file)
+        if isinstance(file_or_image, str):
+            self.core.screenshot(file_or_image)
+        elif file_or_image is not None:
+            self.set_image(file_or_image)
         self.clear(self.background_color)
 
-    def has_key_event(self):
-        return self.core.has_key_event()
 
     def get_event(self, *filter):
+        types = []
+        keys = []
+        combs = []
+        for ent in filter:
+            if ent in [GUI.PRESS, GUI.RELEASE]:
+                types.append(ent)
+            elif isinstance(ent, tuple) or isinstance(ent, list):
+                type, key = ent
+                combs.append((type, key))
+            else:
+                keys.append(ent)
         while True:
             if not self.has_key_event():
                 return False
             self.event = self.get_key_event()
-            if not len(filter) or self.event.type in filter:
+            if not len(combs):
+                if not len(types) or self.event.type in types:
+                    if not len(keys) or self.event.key in keys:
+                        break
+            elif (self.event.type, self.event.key) in combs:
                 break
         return True
 
-    def get_key_event(self):
+    def get_cursor_pos(self):  # ABi
```

Comment:
What's `ABi`?

Reply:
`get_cursor_pos` is an alias of `self.core.get_cursor_pos`, which is in C++ scope, so it's an ABI.

Reply:
hmm, feels a bit redundant, but sounds good

Reply:
Sorry but I'm still confused about this: does `ABi` here stand for [Application binary interface](https://en.wikipedia.org/wiki/Application_binary_interface)? Why is `i` lowercased here?